reports/generic-query-based-data-viewer.md

```diff
@@ -0,0 +1,154 @@
+<!--
+Fill in the WebIDs of the people below.
+Leave this in comments!
+It's possible to have multiple people per role.
+
+Challenge/scenario creator:
+  - https://pieterheyvaert.com/#me
+Solution creator:
+  - https://pieterheyvaert.com/#me
+Report writer:
+  - https://pieterheyvaert.com/#me
+-->
+
+# Generic query-based data viewer
+
+The corresponding challenge is [#114](https://github.com/SolidLabResearch/Challenges/issues/114).
+
+## Problem
+<!--
+You can reuse the pitch of the challenge, but check if you need to make changes.
+For example, it might happen that the approved solution does more than what the original pitch requested.
+-->
+
+When developing apps and adding/removing/editing data in pods, 
+it is useful to be able to easily view what data is actually in your pods. 
+At the moment, there is no generic app that easily allows users to 
+execute queries over multiple pods and inspect the corresponding results.
+
+## Approved solution
+<!--
+Provide information about the approved solution:
+names of tools/libraries created, repos, and so on.
+-->
+
+We developed a [Web app](https://github.com/SolidLabResearch/generic-data-viewer/releases/tag/v1.0.0) that 
+has the following features:
+
+- Before using the app, a developer defines via a config file the SPARQL queries that should be used by the app.
+- Users can log in with their WebID or identity provider.
+- Users select the query based on their name from the sidebar.
+- The app shows the results of the query in a table. 
+- Specific query variables influence how the app shows the results:
+  - If a variable is a link to an image and the variable ends with `_img` then the app shows the actual images.
+  - If a variable is a float and the variable ends with `_float`, the app shows the values as floats.
+- Users can refresh the results of a query via a button.
+
+You find a screencast [here](https://cloud.ilabt.imec.be/index.php/s/AJomCGpLjYbxmCX).
+
+<!--
+Provide a list of important technical decisions and assumptions.
+-->
+We made the following important technological decisions and assumptions:
+
+- The app has the [option](https://github.com/SolidLabResearch/generic-data-viewer/tree/ccccc3aa882bb7c1c76fdafdf5e937ffc491d8af#configuration-file) 
+  to redirect requests from Comunica, which executes the queries, via a proxy. 
+  The app does this to deal with data sources that don't set the CORS headers correctly.
+
+## User flow
+
+<!--
+Describe a concrete user flow with the approved solution.
+Complete the following sections:
+-->
+
+### Actors/actresses
+
+- User of the application.
+
+### Preconditions
+
+- The user has Node.js installed.
+
+### Steps
+
+1. Clone the repository (v1.0.0) via
+   ```shell
+   git clone -b v1.0.0 https://github.com/SolidLabResearch/generic-data-viewer.git
+   ```
+2. Install the dependencies via
+   ```shell
+   npm i 
+   ```
+3. Prepare and start the Community Solid Server with the pods via
```

Comment:
You could also use existing pods for this, no?

Reply:
Yes, but for the steps/demo we want copy/paste instructions that we know work 100% of the time.

Reply:
That's not clear from the title `Steps`, so maybe just rephrase the title to "Steps to reproduce this challenge" and then this comment is resolved and the remainder of my comment could be solved further down (see other comment)